### PR TITLE
Migrate to Go Modules

### LINKED ISSUE #551: Migrate to go modules
Description
===========
Following the 1.9 deprecation in #548 a step forward to be better compatible is to migrate to go modules:
1. Migrate to go modules including examples + kafkatest
2. Some code us gopkg.in and some use another, deprecate all usages of gopkg.in  
3. Update documentation

### PR DESCRIPTION
**Pull request by eran-levy**
fixes #551 

had CLA issues with PR #553 that was created 2 years ago so created this new PR to make sure its clear

### CODE REVIEW

**Comment by eran-levy** [comment]
@edenhill @jeffwidman this is the new PR with all code changes for go modules migration. 

Had issues with PR #553 branch which was 2 years ago - https://github.com/confluentinc/confluent-kafka-go/pull/553#issuecomment-1094095537

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,5 @@
 \#*
 *.prof
 tmp-build
-.DS_Store
\ No newline at end of file
+.DS_Store
+.idea
```
Comment (jeffwidman) [inline .gitignore:6]: How is this related to go modules?

File: kafka/README.md [hunk 2]
```diff
@@ -125,11 +125,6 @@ Manually verify that the examples/ applications work.
 
 Also make sure the examples in README.md work.
 
-Convert any examples using `github.com/confluentinc/confluent-kafka-go/kafka` to use
-`gopkg.in/confluentinc/confluent-kafka-go.v1/kafka` import path.
-
-    $ find examples/ -type f -name *\.go -exec sed -i -e 's|github\.com/confluentinc/confluent-kafka-go/kafka|gopkg\.in/confluentinc/confluent-kafka-go\.v1/kafka|g' {} +
```
Comment (jeffwidman) [inline kafka/README.md:131]: 👍

File: soaktest/go.mod [hunk 3]
```diff
@@ -0,0 +1,16 @@
+module github.com/confluentinc/confluent-kafka-go/soaktest
+
+go 1.13
+
+replace github.com/confluentinc/confluent-kafka-go => ../
+
+require (
+	github.com/DataDog/datadog-go v4.8.3+incompatible
```
Comment (jeffwidman) [inline soaktest/go.mod:8]: Outside of the scope of this PR, but be nice to bump some of these deps to drop the `+incompatible` tags. But in a separate PR IMO.

**Review by jeffwidman — COMMENTED** [review]
I haven't tested locally, but looks reasonable to me. Thanks for keeping this moving.

File: .gitignore (hunk 1 above)
Comment (eran-levy) [reply .gitignore:6]: Not very related just to ignore internal goland IDE and mac folders to eliminate mistakes, I can remove them if its not according to the standards, do you want me to revert this one?

File: soaktest/go.mod (hunk 3 above)
Comment (eran-levy) [reply soaktest/go.mod:8]: Agree, I suggest to do in another PR just to make sure we dont deliver too many changes at once

**Comment by edenhill** [comment]
This is great stuff! Let me know when it is ready to be merged.

**Comment by jeffwidman** [comment]
@edenhill I think that's up to you? It looks ready to me unless you want the `.gitignore` change dropped... I don't feel strongly either way, and besides, it's your project, not mine. 😀

**Comment by eran-levy** [comment]
@jeffwidman @edenhill we can consider this PR as done, otherwise you would like me to make any other changes?

**Review by edenhill — APPROVED** [review]
Great stuff!

**Comment by eran-levy** [comment]
@jeffwidman correct, strange thats it missing, I just see that go.sum is missing only in the root folder and exists in the rest. Will commit the go.sum soon and let you know. 
Thanks that you pointed this out, didnt see it

**Comment by eran-levy** [comment]
@jeffwidman you're correct, but the go.mod under the root folder doesnt include any lib dependencies so that why go.sum is missing. In the rest of the modules such as: examples/ there is a dependency so thats why go.sum exists. 

So seems fine to me, no changes required

**Comment by jeffwidman** [comment]
Oh that makes sense. Thanks!
